hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java

```diff
@@ -214,29 +213,17 @@ private boolean createConfigFiles(String confDirPath, String kernelsDir, String
       String tfLdLibraryPath = tfLibMappingUtil.getTfLdLibraryPath(project);
 
       SparkConfigurationUtil sparkConfigurationUtil = new SparkConfigurationUtil();
-      Map<String, String> extraJavaOptions = new HashMap<>();
+      HashMap<String, String> extraJavaOptions = new HashMap<>();
```

Comment:
@robzor92 Why does this need to be declared as HashMap?

Reply:
I added a HashMap argument containing jobs/jupyter specific jvm options to the setFrameworkProperties arguments. This former Map is passed in there. We anyway instantiate it as a HashMap.

Reply:
@robzor92 It's considered best practice to declare variables and pass arguments with the interface rather than the implementation type. Unless we want to force the caller to use a hashmap or more broadly speaking a specific implementation of an interface, but in this case we don't want that.

Reply:
Okay changing it to map